lib/features/email/domain/usecases/get_email_content_interactor.dart

```diff
@@ -12,31 +13,41 @@ class GetEmailContentInteractor {
 
   GetEmailContentInteractor(this.emailRepository);
 
-  Stream<Either<Failure, Success>> execute(AccountId accountId, EmailId emailId, String? baseDownloadUrl) async* {
+  Stream<Either<Failure, Success>> execute(AccountId accountId, EmailId emailId, String? baseDownloadUrl, {EmailLoaded? emailLoaded}) async* {
     try {
       yield Right<Failure, Success>(GetEmailContentLoading());
-      final email = await emailRepository.getEmailContent(accountId, emailId);
 
-      if (email.emailContentList.isNotEmpty) {
-        final newEmailContents = await emailRepository.transformEmailContent(
-            email.emailContentList,
-            email.allAttachments.listAttachmentsDisplayedInContent,
-            baseDownloadUrl,
-            accountId);
-        final newEmailContentsDisplayed = kIsWeb
-            ? await emailRepository.addTooltipWhenHoverOnLink(newEmailContents)
-            : newEmailContents;
+      if(emailLoaded != null) {
+        await Future.delayed(const Duration(milliseconds: 300));
         yield Right<Failure, Success>(GetEmailContentSuccess(
-            newEmailContents,
-            newEmailContentsDisplayed,
-            email.allAttachments,
-            email));
-      } else if (email.allAttachments.isNotEmpty) {
-        yield Right<Failure, Success>(GetEmailContentSuccess([], [], email.allAttachments, email));
-      } else if (email.headers?.isNotEmpty == true) {
-        yield Right<Failure, Success>(GetEmailContentSuccess([], [], [], email));
+            emailLoaded.emailContents,
+            emailLoaded.emailContentsDisplayed,
+            emailLoaded.attachments,
+            emailLoaded.emailCurrent));
```

Comment:
for this case, why not `dispatchState` directly in controller?

Reply:
dispatchState not work

Reply:
ah, shoudl `consumeState` directly :) srr, I am wrong

Reply:
using `getState`